lib/shrine/plugins/activerecord.rb

```diff
@@ -35,12 +35,12 @@ def included(model)
               attacher.save if attacher.changed?
             end
 
-            model.after_commit on: [:create, :update] do
+            model.after_commit if: :persisted? do
```

Comment:
I'm not so confident this will work with any future version of Rails. Does `#persisted?` surely return `false` when the record is destroyed?

I think it would be safer to register one `after_commit` callback per action:

```rb
[:create, :update].each do |action|
  model.after_commit on: action do
    attacher = send("#{name}_attacher")
    attacher.finalize if attacher.changed?
  end
end
```

Reply:
Yep, overnight and along the day the doubt was itching my mind too :)

Fixed in https://github.com/shrinerb/shrine/pull/398/commits/dcef9108c75de9c49bbf1bd2ea18e94c21f63c62